codeclone/CodeClone01.java

```diff
@@ -0,0 +1,47 @@
+package codeclone;
+
+
+import com.amazonaws.AmazonClientException;
+import com.amazonaws.services.s3.AmazonS3Client;
+import com.amazonaws.services.s3.model.BucketLifecycleConfiguration;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class S3BucketAdapter {
+
+  // move this to brazil config
+  public static final String DEFAULT_BUCKET_NAME = "guru-review-bucket";
+
+  private final AmazonS3Client s3;
+
+  public S3BucketAdapter(AmazonS3Client s3) {
+    this.s3 = s3;
+  }
+
+  public void demoGuru(String jsonBucket) {
+    
+    try {
+      s3.createBucket(array.get(0).getAsString());
+      s3.setBucketLifecycleConfiguration(array.get(0).getAsString(), endpointConfig);
+    } catch (AmazonClientException ex) {
+      String log = "Exception caught trying to create bucket";
+      log.error(log, ex);
+      throw new RuntimeException(logText, ex);
+    }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 27:34, 37:44.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.